bucketeer/build.gradle

```diff
@@ -48,12 +48,12 @@ android {
   }
 
   compileOptions {
-    sourceCompatibility JavaVersion.VERSION_1_8
-    targetCompatibility JavaVersion.VERSION_1_8
+    sourceCompatibility JavaVersion.VERSION_17
+    targetCompatibility JavaVersion.VERSION_17
```

Comment:
Should we keep `VERSION_1_8` here, as this effectively requires SDK consumers to use AGP 8 or greater?

Reply:
Hi @yshrsmz , I tested with an example project, this PR will make breaking changes with A
I found another way to deal with it. I will update the PR soon.

Reply:
Hi @yshrsmz, JVM target may not cause many problems. I tested with some old projects.
But AGP 8.x.x will 
- Make breaking changes for some projects using the legacy library for example `Butterknife` 
- The SDK's consumer needs to use Android Studio FlamingGo (This IDE will create many errors with their project) 

After checking, AGP 7.4.2 is the best choice. (which supports Kotlin 1.8.22).
Other changes in this PR are to fix an error and make our project work well with Android Flamingo.

Reply:
Technically, you should be able to use AGP 8.0.x while keeping source/target compatibility to java11.

AGP 8 requires java17, but it is just for running itself. You can use java17 to compile java11-compatible binary.

But I've tested locally and found one ksp problem: https://github.com/google/ksp/issues/1288

Reply:
```gradle
kotlin {
  jvmToolchain(8)
}

android {
    compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
  }

  kotlinOptions {
    jvmTarget = "1.8"
  }
}
```

This seems to work for me(at least test & assemble works)

Reply:
I tried `jvmToolchain` solution before, and it works.

I think we should go with `7.4.2`.  
It just like the same with `7.3.1` but support the new Kotlin Gradle Plugin version for Kotlin `1.8.22`. The impact is the smallest.
We should wait a little time before jumping for full support AGP 8.x.x 
What do you think @yshrsmz ?